TypeCobol.Analysis.Test/Report/InBulkCallPgm88Set.cbl

```diff
@@ -0,0 +1,63 @@
+﻿       IDENTIFICATION DIVISION.
```

Comment:
Ok there is a bug in the `ValueOrigin.ComputeFrom`: the `MOVE <literal> TO <var>` is not correctly handled so I'll look into that.
In current state, the value 'PGM00000' won't be considered by DFA, the corresponding DefPoint is not in the UseDef list so there is also a second bug either in DFA UseDef building or in CFG. I think that the `PERFORM VARYING` block has to be considered optional by CFG because the stop condition may be true before entering the block and thus, the PERFORM body won't ever be run. So I guess the edge to allow to jump from procedure division start directly to the `CALL` located after `end-perform is missing in CFG.

Reply:
Ok first bug is fixed with a21ba0e874b0ab04c91c383801a41fda18a1ff41. But we still have to look for the missing initial value 'PGM00000' in the results.

Reply:
I agree with this analysis, but one my opinion the only result which is missing here is 
ObjectName=PGM00000;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"PGM00000";SourceText=call 'zcallpgm' using PROGRAM_NAME

So the expected result shall be:
ObjectName=MyPgm;Line=13;Column=17;Path=ProcCall.PROGRAM_NAME<-"MyPgm";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=PGM00000;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"PGM00000";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm2;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm2";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm3;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm3";SourceText=call 'zcallpgm' using PROGRAM_NAME

But using my original framework I get:
ObjectName=MyPgm;Line=13;Column=17;Path=ProcCall.PROGRAM_NAME<-"MyPgm";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm2;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm2";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm3;Line=19;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm3";SourceText=call 'zcallpgm' using PROGRAM_NAME

There is a lack in the DFA calculation I think.

Reply:
Yes, you're right I made a mistake in my test, it should be:
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. ProcCall.
      
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01 PROGRAM_NAME      pic X(08) value 'PGM00000'.
       01 I pic 9(03) value 0.
      
       PROCEDURE DIVISION.
           perform varying I from 1 by 1 until I < 10
               if(I > 50) 
                    move "MyPgm" to PROGRAM_NAME
                end-if  
                call 'zcallpgm' using PROGRAM_NAME
                move "MyPgm2" to PROGRAM_NAME
                if(I > 50) 
                    perform Set-MyPgm3
                end-if
           end-perform
           call 'zcallpgm' using PROGRAM_NAME
           .
       Set-MyPgm3.
           move "MyPgm3" to PROGRAM_NAME
           .
      
       END PROGRAM ProcCall.  
```

In this case, it works almost correctly. I get:
```
ObjectName=MyPgm;Line=14;Column=17;Path=ProcCall.PROGRAM_NAME<-"MyPgm";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm2;Line=14;Column=17;Path=ProcCall.PROGRAM_NAME<-"MyPgm2";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm3;Line=14;Column=17;Path=ProcCall.PROGRAM_NAME<-"MyPgm3";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm2;Line=20;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm2";SourceText=call 'zcallpgm' using PROGRAM_NAME
ObjectName=MyPgm3;Line=20;Column=12;Path=ProcCall.PROGRAM_NAME<-"MyPgm3";SourceText=call 'zcallpgm' using PROGRAM_NAME
```
Only `ObjectName=PGM00000;Line=20;Column=12;Path=ProcCall.PROGRAM_NAME<-"PGM00000";SourceText=call 'zcallpgm' using PROGRAM_NAME` is missing.

Reply:
So the main lack here is that In Cobol variable declaration + definition are not instructions, so my first intuition here is that we must imagine a way to make such (declaration + definition) as dummy instruction in the START blocks, to ensure DFA propagatiion.

Reply:
@mayanje Yes, this could work. Another idea is to modify the `DefautlDataFlowGraphBuilder.GetDefVariables` method so as to report initial definitions of variables on PROCEDURE NODE division node.

Reply:
I that the first initial value is missing for the first call in that second sample code, we should have :
```cob
ObjectName=PGM00000;Line=14;Column=17;Path=ProcCall.PROGRAM_NAME<-"PGM00000";SourceText=call 'zcallpgm' using PROGRAM_NAME
```

Reply:
@fm-117 Yes it seems that the solution is  mix of what you suggest and what I think: collecting all DataDefinition nodes having a value, and Adapt DefautlDataFlowGraphBuilder.GetDefVariables for theses nodes to return the associated VariableSymbol instance. All DataDefnition nodes having a value must appear as Nodes in the START block of the procedure division. DataDefinition node having value are the dummy instructions in this case.

Reply:
done in https://github.com/TypeCobolTeam/TypeCobol/pull/1820